src/games/strategy/common/swing/SwingAction.java

```diff
@@ -0,0 +1,52 @@
+package games.strategy.common.swing;
+
+import java.awt.event.ActionEvent;
+import java.util.function.Consumer;
+import javax.swing.AbstractAction;
+
+
+/**
+ * Builder class for using Lambda expressions to create 'AbstractAction'
+ *
+ * For example, instead of:
+ * <code>
+ *   new AbstractionAction("text") {
+ *         private final long serialVersion = ..
+ *         &#64;Override public void actionPerformed(ActionEvent e ) { doSomething(); }
+ *   }
+ * </code>
+ *
+ * You can rewrite the above using 'SwingAction.of(..)':
+ *
+ * <code>
+ *    SwingAction.of( e -> doSomething() );
+ * </code>
+ */
+public class SwingAction {
+
+  /**
+   * @param name Name for the abstract action, passed along to the AbstractAction constructor.
+   * @param swingAction Lambda java.util.function.Consumer object, accepts one arg and returns void.
+   */
+  public static AbstractAction of(final String name, final Consumer<ActionEvent> swingAction) {
+    return new AbstractAction(name) {
+      private static final long serialVersionUID = 1L;
```

Comment:
Pushed: 44041f9 ,  replaced the "1L" with a generated  random UID.

I had to refresh a bit on serial version UID:
http://stackoverflow.com/questions/285793/what-is-a-serialversionuid-and-why-should-i-use-it

I'm about 95% sure the answer is the different serial version UID does not matter.  It's hard to be certain if we rely on the deserialization of these objects. If we do, then at least we'll have reduced the risk of future unnecessary cross-version ClassCastExceptions.